pkg/collector/corechecks/gpu/model/model.go

```diff
@@ -0,0 +1,62 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package model contains the model for the GPU check, with types shared between the system-probe GPU probe and
+// the gpu core agent check
+package model
```

Comment:
`collector/corechecks` directory is for code that implement the checks, namely the check interface. I would suggest to put shared library code in another package instead. Since the only place using this is `pkg/gpu`, can the model live there also?

Reply:
I followed the structure of other checks (e.g. https://github.com/DataDog/datadog-agent/blob/main/pkg/collector/corechecks/ebpf/probe/oomkill/model/oom_kill_types.go) where the types that will be used to communicate between system-probe and the core agent are defined. I have another PR that implements the agent check, and it requires having the model in a separate package to avoid bringing in the system-probe part of the code.